plugins/newsdownloader.koplugin/main.lua

```diff
@@ -315,4 +315,16 @@ function NewsDownloader:setCustomDownloadDirectory()
     }:chooseDir()
 end
 
+function NewsDownloader:onCloseDocument()
+    local document_path = self.ui.document._document.filename
+    logger.dbg("NewsDownloader: document_path ", document_path)
+    local news_download_dir_without_dot = string.sub(news_download_dir_path,2)
```

Comment:
What's the way news_download_dir looks like? Either use

```lua
-- matches start
-- full path match this way is definitely preferable if at all possible
string.format("^%s", string.sub(news_download_dir_path,2))
```

or

```lua
-- matches end
string.format("%s$", string.sub(news_download_dir_path,2))
```

As opposed to any random occurrence of the string.

Reply:
It doesn't work as we also create subdirs in /news/
eg:
/koreader/news/PCWorld/

Reply:
What part of `string.format("^%s", string.sub(news_download_dir_path,2))` doesn't work?

Reply:
Doesn't. 
news_download_dir_path starts from dot (current koreader work dir, whatever it is, on different devices) : './news/' which is resolved by system to full path. 

Document path is always full path 
/mnt/whatever/... /koreader/news/feed-provider-name/news.html

Reply:
What if you change this assignment to `lfs.currentdir or "."`

https://github.com/koreader/koreader/blob/e8aeb3cfa503017701f2659086d09dc9fededa28/datastorage.lua#L19

Reply:
Alright, I suppose `koreader/news` should be unique enough.

Reply:
I think just `news` is too generic. Is feed_config.lua always present in the directory where news will be downloaded to?

Reply:
Yes it's in parent directory. 
As the dir can be changed by user I'm thinking of a lua regex that ends with .html preceding at some level with news dir. 

Did your commit https://github.com/Frenzie/koreader/commit/58fc7477f8eccfb9e627cdc306521b00fac6ba42 changed the way paths are build? There preceding dot will be resolved to full path?